kernel/src/hil/flash.rs

```diff
@@ -125,3 +127,36 @@ pub trait Client<F: Flash> {
     /// Flash erase complete.
     fn erase_complete(&self, error: Error);
 }
+
+pub trait FlashInfo {
+    fn flash_size(&self) -> u32;
+    fn num_pages(&self) -> u32;
+    fn page_size(&self) -> u32;
+    /// Return 0 if flash does not support locking
+    fn num_lock_units(&self) -> u32;
+    /// Return 0 if flash does not support locking
+    fn lock_unit_size(&self) -> u32;
+    /// Return 0 if flash does not support locking
+    fn pages_per_lock_unit(&self) -> u32;
+    fn page_to_lock_unit(&self, page: u32) -> u32;
+}
+
+pub trait FlashLocking {
```

Comment:
Sorry, I thought this was obvious stuff, but obviously it isn't. Locking lets you make the flash unwriteable. I'll add comments explaining.

I'm hard pressed to think of a Cortex M that doesn't support locking in some form.  Every one I've seen does. It's pretty critical for a very wide range of applications. 

For example, I'd argue that now that we're bringing flash up, one of the first things the kernel should do is lock its own pages. We should therefore make sure that the kernel doesn't share a lock unit with applications.